addons/arsenal/functions/fnc_scanConfig.sqf

```diff
@@ -0,0 +1,195 @@
+/*
+ * Author: Dedmen
+ * Cache an array of all the compatible items for arsenal.
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+*/
+#include "script_component.hpp"
+
+private _cargo = [
+    [[], [], []], // Weapons 0, primary, secondary, handgun
+    [[], [], [], []], // WeaponAccessories 1, optic,side,muzzle,bipod
+    [ ], // Magazines 2
+    [ ], // Headgear 3
+    [ ], // Uniform 4
+    [ ], // Vest 5
+    [ ], // Backpacks 6
+    [ ], // Goggles 7
+    [ ], // NVGs 8
+    [ ], // Binoculars 9
+    [ ], // Map 10
+    [ ], // Compass 11
+    [ ], // Radio slot 12
+    [ ], // Watch slot  13
+    [ ], // Comms slot 14
+    [ ], // WeaponThrow 15
+    [ ], // WeaponPut 16
+    [ ] // InventoryItems 17
+];
+
+private _configCfgWeapons = configFile >> "CfgWeapons"; //Save this lookup in variable for perf improvement
+private _magazineGroups = [[],[]] call CBA_fnc_hashCreate;
+
+{
+    private _configItemInfo = _x >> "ItemInfo";
+    private _simulationType = getText (_x >> "simulation");
+    private _className = configName _x;
+
+    switch true do {
+        /* Weapon acc */
+        case (
+                isClass (_configItemInfo) &&
+                {(getNumber (_configItemInfo >> "type")) in [101, 201, 301, 302]} &&
+                {!(configName _x isKindOf ["CBA_MiscItem", (_configCfgWeapons)])}
+            ): {
+
+            switch (getNumber (_configItemInfo >> "type")) do {
+                case 201: {
+                    (_cargo select 1) select 0 pushBackUnique _className;
+                };
+                case 301: {
+                    (_cargo select 1) select 1 pushBackUnique _className;
+                };
+                case 101: {
+                    (_cargo select 1) select 2 pushBackUnique _className;
+                };
+                case 302: {
+                    (_cargo select 1) select 3 pushBackUnique _className;
+                };
+            };
+
+        };
+        /* Headgear */
+        case (isClass (_configItemInfo) && {getNumber (_configItemInfo >> "type") == 605}): {
```

Comment:
you can also do
```
if (cond) then {
case:
case: 
case:
}
```
I never tried that. But I don't see a reason why that wouldn't work.